action.yml

```
@@ -45,6 +47,7 @@ runs:
             -v /var/run/docker.sock:/var/run/docker.sock:ro \
             -v ~/.docker:/root/.docker \
             -v ${{ github.workspace }}:/data \
+            -e CAS_API_KEY="${{ input.cas_token }}" \
```

```suggestion
            -e CAS_API_KEY="$CAS_API_KEY" \
```

By doing this, we can drop the input and it would be invoked by:
```
...
      - name: Publish ${{ matrix.architecture }} builder
        uses: ./
        with:
          args: ...
        env:
          CAS_TOKEN: ${{ secrets.CAS_TOKEN }}
...
```

---

Another option is to use --env-file and point to the action env file, that way it's more dynamic and we don't have to adjust in the future

---

`--env-file "$GITHUB_ENV"`

---

Yes, I can write the inputs.cas_token into an env-file, but what is the benefit? We should also ditch the args and make real parameters

---

`--env-file "$GITHUB_ENV"` that work?

---

I have not tested it, but the documentation states that it should work <https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#environment-files>
`$GITHUB_ENV` points to a file that contains all default and passed vars in a step

---

Tested it, it does not work...